src/tools/pub/dependencies.md

```diff
@@ -442,6 +439,32 @@ To identify dependencies in your app or package that
 aren't on the latest stable versions,
 use [`dart pub outdated`][].
 
+### Use tight dev_dependencies
+
+Dependencies can be loose, specifying dependency at the highest version digit
+like `^2.0.0`, or tight, specifying dependency at the lowest version digit like
+`^1.2.4`.
```

Comment:
I don't think we should necessarily define this in terms of which "version digit" as that's not a term we've used on the site. If someone is reading the best practices I think we can assume prerequisite knowledge of what each portion of the version specifier means, so let's specify it in terms of that (major, minor, patch).

Also, as mentioned above, it's best to refer to the constraints being tight or loose.

So in the case of a tight version constraint, that means the lower bound being the newest compatible version.

Reply:
I'd skip the "loose" definition here if the best practice is just "tighten", define a "tight" constraint ("_that means the lower bound being the newest compatible version._"), and like Parker said [below](https://github.com/dart-lang/site-www/pull/5431/files#r1449864165), add the reasoning and definition of the reasoning to the opening description of the best practice, e.g.

    Specify dev dependencies at their lower bound, or newest compatible version,
    to ... (what does this achieve / avoid?)  
    OR
    because ... (something something transitive concerns?)